### DISCUSSION TITLE
Prognose um eine Stunde vorgeschoben

### DISCUSSION THREAD

**Discussion by iReckyy** [wiggal/GEN24_Ladesteuerung#71] [Q&A] [answered]
Hallo,

ich nutze die Ladesteuerung um die 70%-Regelung zu umgehen und bin grundsätzlich begeistert von diesem Programm.

Jetzt waren endlich mal vollständig sonnige Tage, sodass man die Prognose gut mit dem "Ist-Ertrag" abgleichen kann.

Die Prognose beziehe ich aktuell von Solarprognose.de
Azimuth- und Dachneiguns-Winkel wurden penibel ermittelt.

Das Problem ist nun, dass die Prognose um eine Stunde nach vorne verschoben zu sein scheint.
Ich habe alle Uhren der verschiedenen Systeme gecheckt, die passen alle.

Habe ich den Azimuth-Winkel doch zu niedrig von Süd aus eingetragen?

Es ist eine SW-Anlage.

Ich hänge mal ein Bild an, zur veranschaulichung.

<img width="1680" alt="Bildschirmfoto 2024-06-25 um 18 25 16" src="https://github.com/wiggal/GEN24_Ladesteuerung/assets/173826530/45b7f880-5d4a-47d6-8a32-a0d161b33c89">

**Comment by wiggal** [comment]
Sieh dir mal den Post an:
https://github.com/wiggal/GEN24_Ladesteuerung/issues/65#issuecomment-2119732113
Den Passus unter EDIT:
Vielleicht löst das dein Problem

**Reply by wiggal** [reply]
könnte aber auch das Gegenteil bewirken. evtl. die Konfig etwas nach Westen drehen.

**Reply by iReckyy** [reply]
Das war nicht angehakt, hatte ich eben mal gemacht.
Was es bewirkt weiß ich noch nicht.

Interessant ist, dass bei der Benutzerzeitzone bei Europa/Berlin UTC+2 steht.

Ich denke mal das hat mit der Sommerzeit zu tun.

Nach Westen drehen wäre für mich die letzte Option, ist halt ein wenig Pfusch.


Bei Solarprognose.de ist die Vorhersage allerdings auch um die Stunde früher als der Ertrag.

**Reply by wiggal** [reply]
Bei mir steht dort auch UTC+2.
Ich verwende Solcast, bei mir war die Prognose immer ca. eine Stunde zu spät.
Ich hab die Ausrichtung jetzt ein bisschen nach Osten gedreht, und jetzt passt es etwas besser, für die Berechnungen reicht es.
Hier vorher und nachher:
![grafik](https://github.com/wiggal/GEN24_Ladesteuerung/assets/69506553/adbfb8a4-8ba3-4b56-87a9-db22235e10de)
![grafik](https://github.com/wiggal/GEN24_Ladesteuerung/assets/69506553/d27ce5ce-9b66-4437-94a0-8c277587e8d2)

Übrigens, wenn du ein Problem hast, dass der Akku zu schnell voll ist, die HTTP-Version ist hier genauer und damit besser.

**Comment by iReckyy** [comment]
Habe jetzt mal auf die http Version umgesetllt, die läuft garnicht -.-


Traceback (most recent call last):
File "/home/GEN24/http_SymoGen24Controller2.py", line 22, in
Parameter = getParameter(argv)
^^^^^^^^^^^^^^^^^^
File "/home/GEN24/FUNCTIONS/fun_Ladewert.py", line 276, in getParameter
Prog_Steuer_code = int(Prog_Steuer_code_tmp['Steuerung'])
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'list'

**Reply by wiggal** [reply]
ups,
die Datei Prog_Steuerung.json auf Github ist falsch. 
Geh mal ins Register Settigs und wähle "unverändert lassen" aus und speichern.
Dann müsste er die Datei richtig schreiben, sollte so aussehen:
```
{
    "Steuerung": "0"
}
```

**Reply by wiggal** [reply]
Oder jetzt die Datei neu von Github holen, habe sie bereinigt

**Reply by iReckyy** [reply]
Teste ich, melde mich gleich.

**Reply by iReckyy** [reply]
Ist nicht wirklich besser geworden.



Traceback (most recent call last):
  File "/home/GEN24/http_SymoGen24Controller2.py", line 95, in <module>
    API = get_API()
          ^^^^^^^^^
  File "/home/GEN24/FUNCTIONS/fun_API.py", line 23, in get_API
    API['aktuelleEinspeisung'] = int(data['Body']['Data']['16711680']['channels']['SMARTMETER_POWERAPPARENT_MEAN_SUM_F64'])
                                     ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
KeyError: '16711680'

**Reply by wiggal** [reply]
Was hast du denn für einen GEN24?
Bei dir scheint es den Schüssel '16711680' nicht zu geben, komisch.
Ich hab den Schlüssel mal auf '16252928' geändert, da der schon beim Logging verwendet wurde.
Du müsstest jetzt die Datei 'FUNCTIONS/fun_API.py' neu ziehen.

mal sehen

EDIT:
oder welche Firmware?

**Reply by iReckyy** [reply]
Gen24 8.0, gestern auf die neuste Firmware geupdatet - 1.32.5-1

Jetzt sagt er falsche Zugansgdaten.

Das logging/debugging klappt auch nciht richtig, kann nur die Date lesen, über die Website zeigt er nichts an.


Per Modbus hat er jetzt geschrieben und es auch geloggt.

Jetzt loggt er auch beim http-Zugriff, allerdings sagt er falsche Zugangsdaten... ich verstehe nicht warum, die passen eigentlich.
Oder geht das nicht über den Technician User?



Daten wurden in die SQLite-Datei gespeichert!
************* ENDE: 2024-06-26 16:46:02.868248 *************

[Fronius] Login fehlgeschlagen .. falsche Zugangsdaten?

**Reply by wiggal** [reply]
läuft das Skript http_Symio... ohne Fehler durch?

**Reply by wiggal** [reply]
doch muss auch über den Technician funktionieren. technican, klein geschrieben?

**Reply by iReckyy** [reply]
Wo sehe ich ob das Script ohne Fehler durch läuft?

Also am Gen24 Tab kann ich mich mit Technician und dem Passwort anmelden.
Wenn ich das Script starte sagt er falsche Zugangsdaten...

**Reply by wiggal** [reply]
Hast du die Zugangsdaten/Kennwort ([gen24] password)  in der config.ini eingegeben?

Fehlerfrei = wenn du keine 'Traceback (most recent call last)' Meldungen bekommen hast

**Reply by iReckyy** [reply]
Dann läuft das Script durch.

Also ich habe gerade festgestellt, dass ich das Kennwort nicht in Hochkommas hatte, die Anmeldung schlägt dennoch fehl....

Über Modbus läuft es normal.

**Reply by iReckyy** [reply]
Ok..... ich bin doof.

technician muss klein geschrieben werden und das Kennwort in Hochkommas.

Es läuft jetzt. 

Vielen Dank für die Hilfe.



Habe jetzt mal auf die Forecast.Solar (Weatherprovider2) API umgestellt und werde die Prognose beobachten.

Mit dem Haken bei Solaprognose gesetzt kam heute irgendwie was ganz komsiches raus.

**Reply by wiggal** [reply]
OK, dann scheint sich nur der eine Schlüssel geändert zu haben.

Zur Prognose, ich bin mit solcast.com recht zufrieden.

**Reply by iReckyy** [reply]
Ok, kann ich denn da meine Winkel genau eintragen und es passt dann zeitlich?

In der Config habe ich noch einen Parameter gesehen um die Zeiten zu schieben.

**Reply by wiggal** [reply]
Ja die Werte hatte ich eingetragen wie sie sind, eine unbedeutende Verschiebung hab ich in den letzten Tagen gemacht, aber ist nur Spielerei.
Der Parameter +1 in der config passt für Zeitzone Berlin.
Die Abfrage Häufigkeit ist halt auf 5x am Tag limitiert, ich frag nur 4x ab, damit ich immer einen Puffer hab.

**Reply by iReckyy** [reply]
Ok, aber wenn da auch eine Verschiebung auftreten sollte, könnte ich das mit dem Azimuthwinkel anpassen, oder kann ich auch bspw. ne halbe Stunde eintragen, oder 1,5 Stunden?

**Reply by wiggal** [reply]
Zeitzone geht nur Ganzzahl

**Comment by wiggal** [comment]
übrigens wenn du Solcast testen willst, einfach in dem Block den Dateinamen unter weatherfile ändern, dann kannst du die Werte vergleichen.

**Reply by iReckyy** [reply]
Stimmt, danke für den Tipp.

**Comment by iReckyy** [comment]
Ich weiß nicht warum, aber nachdem ich die Settings bei Solarprognose.de wieder wie vorher eingestellt habe passt es jetzt sehr gut.
<img width="1680" alt="Bildschirmfoto 2024-06-28 um 19 12 14" src="https://github.com/wiggal/GEN24_Ladesteuerung/assets/173826530/cf6ce6d3-ebfe-41c2-90b1-28da66d553f9">

**Reply by wiggal** [reply]
Na ist doch toll, man steckt halt oft nicht drin.
Ich sag immer: "So ein Computer ist auch nur ein Mensch" ;-)

**Comment by iReckyy** [comment]
Ich weiß nicht warum, aber bei Solarprognose ist die Vorhersage wieder eine Stunde zu früh...

Ist es evtl. möglich dort wie bei Solcast.com einen Stunden-Offset einzubauen?

**Reply by wiggal** [reply]
> Ich weiß nicht warum, aber bei Solarprognose ist die Vorhersage wieder eine Stunde zu früh...

Was hast du den bei Solarprognose eingestellt?
So habe ich es eingestellt:
![grafik](https://github.com/user-attachments/assets/ba4edd0a-d345-4d03-84eb-926016860956)
==============
![grafik](https://github.com/user-attachments/assets/421ecd27-ef4f-493b-8db3-767b8be7e492)

> Ist es evtl. möglich dort wie bei Solcast.com einen Stunden-Offset einzubauen?

Eigentlich nicht, da dies bei Solarprognose nicht vorgesehen ist.

**Reply by iReckyy** [reply]
Fast genau so.
.
.
.
![Benutzerprofil](https://github.com/user-attachments/assets/27460f31-3fb3-407c-acd2-058deb933d1a)
![Benutzereinstellungen](https://github.com/user-attachments/assets/fda451e6-8a86-4d93-bcc5-752b811cfe8b)

**Reply by iReckyy** [reply]
Es ist halt genau eine Stunde zu früh, auch im Vergleich zu Solcast.com, bei identischen Winkeln.
Habe Solarprognose.de mal angeschrieben und gefragt ob sie was dazu sagen können.
.
.
.
.
<img width="1680" alt="Bildschirmfoto 2024-07-16 um 11 54 34" src="https://github.com/user-attachments/assets/863a1774-67b8-448e-8f26-aead4474c8ce">

**Reply by wiggal** [reply, marked as answer]
Schon sehr komisch bei mir ist es mit denselben Einstellungen eher eine Stunde zu spät.
OK, ich baue die Verschiebung bei Gelegenheit ein.
![grafik](https://github.com/user-attachments/assets/488240dd-850d-4517-b4a4-07fff0417034)

**Reply by iReckyy** [reply]
Top, das freut mich sehr.

**Reply by wiggal** [reply]
OK, habe es eingebaut, kannst testen.

**Reply by iReckyy** [reply]
Ui, das ging schnell.

**Reply by iReckyy** [reply]
Habe das Update drin, leider aufwändig, wegen der config.ini, siehe anderer Beitrag.

Es sieht super aus. Die Prognosen haben den höchsten Wert jetzt da wo er hin gehört.
Werde berichten ob es jetzt genau passt
